Sources/SwiftParser/Expressions.swift

```diff
@@ -2045,6 +2050,33 @@ extension Parser.Lookahead {
   }
 }
 
+// MARK: Do-Catch Expressions
+
+extension Parser {
+  /// Parse a do expression.
+  mutating func parseDoExpression(doHandle: RecoveryConsumptionHandle) -> RawDoExprSyntax {
+    assert(experimentalFeatures.contains(.doExpressions))
```

Comment:
I would make this a precondition based on 

https://github.com/apple/swift-syntax/blob/7a643a740411befe13c797ccadbbf372e19bda8e/Sources/SwiftSyntax/Assert.swift#L13-L22

Reply:
Heh, I'm too used to C++ 😅